apis/v1beta1/config.go

```diff
@@ -133,6 +139,53 @@ type Config struct {
 	Service    Service    `json:"service" yaml:"service"`
 }
 
+// getPortsForComponentTypes gets the ports for the given ComponentType(s).
+// NOTE FOR REVIEWERS: we could also do this by introducing something in components that is aware of which retriever
+// is being used and pass the right config in where necessary...
+func (c *Config) getPortsForComponentTypes(logger logr.Logger, componentType ...ComponentType) ([]corev1.ServicePort, error) {
+	var ports []corev1.ServicePort
+	enabledComponents := c.GetEnabledComponents()
+	for _, cType := range componentType {
+		var retriever components.ParserRetriever
+		var cfg AnyConfig
+		switch cType {
+		case ComponentTypeReceiver:
+			retriever = receivers.ReceiverFor
+			cfg = c.Receivers
+		case ComponentTypeExporter:
+			retriever = exporters.ParserFor
+			cfg = c.Exporters
+		case ComponentTypeProcessor:
+			break
+		}
+		for componentName := range enabledComponents[cType] {
+			// NOTE FOR REVIEWERS:
```

Comment:
Seeking opinions here

Reply:
Passing in the config is fine imo, but I'd make the parsers themselves aware of their own component type in a follow up. They kind of know anyway right now, just with the double underscore prefix.